unitTests/ui/ResultLinkTest.ts

```diff
@@ -66,8 +78,19 @@ export function ResultLinkTest() {
 
     it('should contain the clickUri if the result has no title', () => {
       fakeResult.title = undefined;
-      test = Mock.advancedResultComponentSetup<ResultLink>(ResultLink, fakeResult, undefined);
-      expect(test.cmp.element.innerHTML).toEqual(fakeResult.clickUri);
+      fakeResult.clickUri = 'https://www.google.com?q=hello&geo=world';
+      initResultLink();
+
+      const encodedUri = fakeResult.clickUri.replace('&', '&amp;');
+      expect(test.cmp.element.innerHTML).toEqual(encodedUri);
+    });
+
+    it('when the title is empty and the clickuri contains a html element, it does not render the element', () => {
+      fakeResult.title = '';
+      fakeResult.clickUri = '<IMG SRC=/ onerror="alert(String.fromCharCode(88,83,83))"></img>';
```

Comment:
We should test for `cmp.element.innerHTML` to be the escaped version of your test string.